### PR TITLE
Fix piesno type

### PR DESCRIPTION
**Pull request by nilgoyette**
I found a problem with piesno. The line 

    np.sum(data**2, axis=-1, dtype=np.float32)

doesn't do what you think it does. For a dwi for example, the `dtype` is `int16` and some values^2 are greater then 2^15.

The problem is that `data**2` is still in `int16`, then it's converted to a `float32` so this image is curiously bounded with the `int16` bounds :)

Moreover, I changed `axis=-1` to `axis=2` because we always know the right axis... it's 2.

### CODE REVIEW

**Comment by arokem** [comment]
Good catch. Would you mind writing a test that fails without this fix, so that we can avoid reintroducing it? I'd say keep `axis=-1` for the cases where I feed in 2D arrays as data.

**Comment by nilgoyette** [comment]
I'll see what I can do.

> for the cases where I feed in 2D arrays as data.

    # This method works on a 2D array with repetitions as the third dimension,
    # so process the dataset slice by slice.
    if data.ndim < 3:
        e_s = "This function only works on datasets of at least 3 dimensions."
        raise ValueError(e_s)

My change is in `_piesno_3D` which will always receive 3D data.

**Comment by arokem** [comment]
Duh, of course. Thanks!

**Comment by nilgoyette** [comment]
I don't know what to do with those 10 python 3.5 errors, they are all "ok".

    Expected:
        array([ 1. ,  1.5,  2.5,  3.5,  4.5,  5. ])
    Got:
        array([1. , 1.5, 2.5, 3.5, 4.5, 5. ])

Do you ignore them?

**Comment by nilgoyette** [comment]
I tried to use the normal `test_piesno` image for the test but I always had the same answers. I created my own and took `N=2` to see a difference.

File: dipy/denoise/nlmeans.py [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-
-# warnings.simplefilter('always', DeprecationWarning)
-# warn(DeprecationWarning("Module 'dipy.denoise.nlmeans' is deprecated,"
-#                         " use module 'dipy.denoise.non_local_means' instead"))
+from dipy.denoise.noise_estimate import estimate_sigma, piesno
 
 
 def nlmeans(arr, sigma, mask=None, patch_radius=1, block_radius=5,
-            rician=True, num_threads=None):
+            rician=False, num_threads=None):
```
Comment (arokem) [inline dipy/denoise/nlmeans.py]: Why did you change this default?

File: dipy/denoise/nlmeans.py [hunk 2]
```diff
@@ -67,15 +59,49 @@ def nlmeans(arr, sigma, mask=None, patch_radius=1, block_radius=5,
...
                                               num_threads).astype(arr.dtype)
+            print(datetime.now() - start)
 
         return denoised_arr
 
     else:
         raise ValueError("Only 3D or 4D array are supported!", arr.shape)
+
+
+def main():
```
Comment (arokem) [inline dipy/denoise/nlmeans.py]: Please remove this code. I think that is causing the current failures.

File: dipy/denoise/nlmeans.py [hunk 3]
```diff
@@ -67,15 +59,49 @@ def nlmeans(arr, sigma, mask=None, patch_radius=1, block_radius=5,
             sigma = np.ones(arr.shape, dtype=np.float64) * sigma
 
         for i in range(arr.shape[-1]):
+            start = datetime.now()
```
Comment (arokem) [inline dipy/denoise/nlmeans.py]: Please remove the timing code from here.

File: dipy/denoise/nlmeans.py [hunk 4]
```diff
@@ -67,15 +59,49 @@ def nlmeans(arr, sigma, mask=None, patch_radius=1, block_radius=5,
...
             denoised_arr[..., i] = nlmeans_3d(arr[..., i],
                                               mask,
                                               sigma[..., i],
                                               patch_radius,
                                               block_radius,
                                               rician,
                                               num_threads).astype(arr.dtype)
+            print(datetime.now() - start)
```
Comment (arokem) [inline dipy/denoise/nlmeans.py]: And this as well

**Comment by nilgoyette** [comment]
**Damn, sorry, my bad.** These change are unrelated to the current topic. I did a `commit -a` without thinking. Will remove tomorrow at work.

**Comment by arokem** [comment]
Hi @nilgoyette : the failures you see on the last bot are probably unrelated to your work here. This bot tests new releases of numpy/scipy before they go live, so that we can fix things in advance. We'll need to fix these on a separate PR, before we can merge yours (which I believe is fine).

**Comment by arokem** [comment]
OK - now that #1377 is merged, this failure should go away. 

Could you please rebase your branch on top of master?

**Comment by nilgoyette** [comment]
I did a rebase/push as suggested but it doesn't seem to start the CI. Is there a way to force a tests-run?

**Comment by arokem** [comment]
Maybe try again? I just merged #1378, so if you rebase again, that might sort this one out.

**Comment by arokem** [comment]
Off we go! Thanks!
